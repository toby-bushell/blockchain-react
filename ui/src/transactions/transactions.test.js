import React from 'react';
import renderer from 'react-test-renderer';
import Transactions from './transactions';

it('renders correctly', () => {
  const transactionsDummy = {
    sent: [
      {
        fromAddress: 'XIbniDmR0IBL4ohswo2S',
        toAddress: '21421421',
        amount: '10'
      },
      {
        fromAddress: 'XIbniDmR0IBL4ohswo2S',
        toAddress: 'tobbyyy',
        amount: '200'
      }
    ],
    received: [
      {
        fromAddress: null,
        toAddress: 'XIbniDmR0IBL4ohswo2S',
        amount: 100
      },
      {
        fromAddress: null,
        toAddress: 'XIbniDmR0IBL4ohswo2S',
        amount: 100
      },
      {
        fromAddress: null,
        toAddress: 'XIbniDmR0IBL4ohswo2S',
        amount: 100
      }
    ]
  };
  const tree = renderer.create(<Transactions transactions={transactionsDummy} />).toJSON();

  expect(tree).toMatchSnapshot();
});
