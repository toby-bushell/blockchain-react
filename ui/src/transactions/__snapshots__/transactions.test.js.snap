// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="sc-bwzfXH CzfMt"
  >
    <h3>
      Received
    </h3>
    <table
      className="sc-bdVaJa hQLGxP"
    >
      <thead>
        <tr>
          <th>
            To Address
          </th>
          <th>
            From Address
          </th>
          <th>
            Amount (tbc)
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            XIbniDmR0IBL4ohswo2S
          </td>
          <td>
            (mining reward)
          </td>
          <td>
            100
          </td>
        </tr>
        <tr>
          <td>
            XIbniDmR0IBL4ohswo2S
          </td>
          <td>
            (mining reward)
          </td>
          <td>
            100
          </td>
        </tr>
        <tr>
          <td>
            XIbniDmR0IBL4ohswo2S
          </td>
          <td>
            (mining reward)
          </td>
          <td>
            100
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="sc-bwzfXH CzfMt"
  >
    <h3>
      Sent
    </h3>
    <table
      className="sc-bdVaJa hQLGxP"
    >
      <thead>
        <tr>
          <th>
            To Address
          </th>
          <th>
            From Address
          </th>
          <th>
            Amount (tbc)
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            21421421
          </td>
          <td>
            XIbniDmR0IBL4ohswo2S
          </td>
          <td>
            10
          </td>
        </tr>
        <tr>
          <td>
            tobbyyy
          </td>
          <td>
            XIbniDmR0IBL4ohswo2S
          </td>
          <td>
            200
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
